### ISSUE TITLE
Should not lint on function signature variable names

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#17]
## box.linters version

0.0.0.9006

## Source code to lint

```R
some_function <- function(x, y) {
  x
  sum(x)
}

result <- some_function(4)
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Expected result

Should not lint on variable names declared in function signature (SYMBOL_FORMAL). Used or unused.
